client/src/components/DateTimePickerRanges.js

```diff
@@ -0,0 +1,83 @@
+import "date-fns";
+import React, { useState } from "react";
+import { Grid } from "@material-ui/core/";
+import { DateTimePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
+import DateFnsUtils from "@date-io/date-fns";
+import { isSameDay } from "date-fns";
+
+const allowedDropInDates = (ranges) => (day) => {
+  const dayDate = new Date(day);
+
+  return ranges.every(({ start, end }) => {
+    if (isSameDay(new Date(day), new Date(start))) return false;
+    return dayDate <= new Date(start) || dayDate > new Date(end);
+  });
+};
+
+const allowedDropOffDates = (ranges, leftDate) => (day) => {
+  const dayDate = new Date(day);
+  if (isSameDay(leftDate, dayDate)) return false;
+  if (leftDate >= dayDate) return true;
+
+  for (let i = 0; i < ranges.length; i++) {
+    if (
+      dayDate > new Date(ranges[i].start) &&
+      dayDate < new Date(ranges[i].end)
+    ) {
+      if (
+        leftDate >= new Date(ranges[i].start) &&
+        leftDate <= new Date(ranges[i].end)
+      ) {
+        return false;
+      }
+    }
+  }
+  return true;
+};
+
```

Comment:
Might want to double check the "drop off" date because it's still possible to choose a time in the past when both drop in and drop off are on the same date